cmd/generate/generate.go

```diff
@@ -347,6 +352,45 @@ func main() {
 		if err := os.WriteFile(filepath.Join(output, "images.yaml"), mapping, fs.ModePerm); err != nil {
 			log.Fatal("Write images mapping file ", err)
 		}
+	} else if generators == GenerateMustGatherDockerfileOption {
+		var rpmsLockTemplate *embed.FS
+		templateName = mustGatherDockerfileTemplateName
+		metadata, err := project.ReadMetadataFile(projectFilePath)
+		if err != nil {
+			if !errors.Is(err, os.ErrNotExist) {
+				log.Fatal("Failed to read project metadata file: ", err)
+			}
+			log.Println("File ", projectFilePath, " not found")
+			metadata = nil
+		}
+
+		projectName := mustGatherDockerfileTemplateName
+		projectDashCaseWithSep := projectName + "-"
+
+		d := map[string]interface{}{
+			"main":             projectName,
+			"must_gather_base": mustGatherBaseImage,
+			"version":          metadata.Project.Version,
+			"project":          capitalize(projectName),
+			"project_dashcase": projectDashCaseWithSep,
+		}
+		out := filepath.Join(output, dockerfilesDir, filepath.Base(projectName))
+		saveDockerfile(d, DockerfileMustGatherTemplate, out, "")
+		rpmsLockTemplate = &RPMsLockTemplate
+
+		t, err := template.ParseFS(rpmsLockTemplate, "rpms.lock.yaml")
+		if err != nil {
+			log.Fatal("Failed creating RPM template ", err)
+		}
+
+		bf := &buffer.Buffer{}
+		if err := t.Execute(bf, nil); err != nil {
+			log.Fatal("Failed to execute RPM template", err)
+		}
+
+		if err := os.WriteFile(filepath.Join(rootDir, cachi2DefaultRPMsLockFilePath), bf.Bytes(), 0644); err != nil {
+			log.Fatal("Failed to write RPM lock file", err)
+		}
```

Comment:
nit: extract a function for this, as we do the same in https://github.com/openshift-knative/hack/blob/2b18900c0f643f85ab98f52b98aef3b50ee3c682/cmd/generate/generate.go#L323-L335 too

Reply:
I'll do it in follow up